lib/web_ui/dev/test_runner.dart

```diff
@@ -279,8 +308,42 @@ class TestCommand extends Command<bool> with ArgUtils {
   bool get isFirefox => browser == 'firefox';
 
   /// Whether [browser] is set to "safari".
-  bool get isSafariOnMacOS => browser == 'safari'
-      && io.Platform.isMacOS;
+  bool get isSafariOnMacOS => browser == 'safari' && io.Platform.isMacOS;
+
+  /// Due to efficiancy constraints, Chrome integration tests only run on
+  /// Linux on LUCI.
+  ///
+  /// They run on all platforms for local.
+  bool get isChromeIntegrationTestAvailable =>
+      (isChrome && isLuci && io.Platform.isLinux) || (isChrome && !isLuci);
+
+  /// Due to efficiancy constraints, Firefox integration tests only run on
+  /// Linux on LUCI.
+  ///
+  /// For now Firefox integration tests only run on Linux and Mac on local.
+  ///
+  // TODO: https://github.com/flutter/flutter/issues/63832
+  bool get isFirefoxIntegrationTestAvailable =>
+      (isFirefox && isLuci && io.Platform.isLinux) ||
+      (isFirefox && !isLuci && !io.Platform.isWindows);
+
+  /// Latest versions of Safari Desktop are only available on MacOS.
+  ///
+  /// Integration testing on LUCI is not supported at the moment.
+  // TODO: https://github.com/flutter/flutter/issues/63710
+  bool get isSafariIntegrationTestAvailable =>
+      (isSafariOnMacOS && isLuci && io.Platform.isLinux) ||
+      (isSafariOnMacOS && !isLuci && !io.Platform.isWindows);
+
+  /// Due to various factors integration tests might be missing on a given,
+  /// platform and given environment.
+  /// See: [isChromeIntegrationTestAvailable]
+  /// See: [isSafariIntegrationTestAvailable]
+  /// See: [isFirefoxIntegrationTestAvailable]
+  bool get isIntegrationTestsAvailable =>
+      isChrome && isChromeIntegrationTestAvailable ||
+      isFirefox && isFirefoxIntegrationTestAvailable ||
+      isSafariOnMacOS && isSafariIntegrationTestAvailable;
```

Comment:
`isChrome`, `isFirefox`, and `isSafariOnMacOS` are redundant. They are already checked by the `is*IntegrationTestAvailable` getters.

Reply:
This is faster. Let me explain. `isChrome`, `isFirefox`,  are the first conditions on an `and` statement, so if they are false, second conditions `isChromeIntegrationTestAvailable` or `isFirefoxIntegrationTestAvailable` (which have other `or` statements) are not executed, so the current implementation saves cycles.

On the other hand this is not on critical path so it doesn't matter. I can remove them